File: wallet/src/main/java/chatox/wallet/api/response/UserResponse.java
package chatox.wallet.api.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponse {
    private String id;
    private String slug;
    private String firstName;
    private String lastName;
    private ZonedDateTime createdAt;
    private ZonedDateTime dateOfBirth;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String email;

    private UploadResponse<ImageMetadata> avatar;
    private String externalAvatarUri;
}

File: user/src/main/kotlin/chatox/user/api/response/UserResponse.kt
package chatox.user.api.response

import chatox.user.domain.ImageUploadMetadata
import com.fasterxml.jackson.annotation.JsonInclude
import java.time.ZonedDateTime

data class UserResponse(
        val id: String,
        @field:JsonInclude(JsonInclude.Include.NON_NULL)
        val accountId: String? = null,
        val slug: String?,
        val bio: String?,
        val firstName: String,
        val lastName: String?,
        val createdAt: ZonedDateTime,
        val lastSeen: ZonedDateTime,
        val dateOfBirth: ZonedDateTime?,
        val online: Boolean,
        val anonymous: Boolean,
        @field:JsonInclude(JsonInclude.Include.NON_NULL)
        val email: String? = null,
        val avatar: UploadResponse<ImageUploadMetadata>?
)
